scss/_utilities_layout.scss

```diff
@@ -3,8 +3,10 @@
     @extend %fixed-width-container;
 
     & &,
-    .row &,
-    & .row {
+    [class^='row'] &,
+    [class*=' row'] &,
+    & [class^='row'],
+    & [class*=' row'] {
```

Comment:
This works and is quite clever way to implement it, but is also a bit unclear (if you don't understand the intentions).

The way we do it in other places (when there are many variants to cover), is to have one standard class selector for the "regular" component `.row`, and a second one for all variants `[class*='row--']`.

So in this case it would look like:
```scss
& &,
.row &,
[class*='row--'] &,
& .row,
& [class*='row--'] {
```

Reply:
On the other hand, just a thought.

Would it work, if we just move this reset inside the `%fixed-width-container`? It technically could be just `& &` inside there, and whatever will extend it, would automatically work? This should generate all the combinations without us explicitly defining them (if I understand it correctly).

This could simplify this and avoid such issues in future if we introduce more variants.

Reply:
Wouldn't `& &` be the same as `.u-fixed-width .u-fixed-width`?

Reply:
Yes, if we keep it here, in definition of `u-fixed-width`. But if we move `& &` inside the definition of `%fixed-width-container`, I believe it would generate the combinations for everything that extends `%fixed-width-container` which would be both `u-fixed-width` and all combinations of rows.

Worth trying I guess.

Reply:
I see what you mean now, let's give it go.